docs/gallery/advanced_io/plot_editing.py

```diff
@@ -0,0 +1,178 @@
+# arbitrarily long in that dimension. Chunking is required for datasets with flexible
+# shapes. Setting ``maxshape`` automatically sets chunking to ``True``, if not specified.
+#
+# First, let's create an NWB file with a dataset with a flexible shape:
+
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+nwbfile = NWBFile(
+    session_description="my first synthetic recording",
+    identifier="EXAMPLE_ID",
+    session_start_time=datetime.now(tzlocal()),
+    session_id="LONELYMTN",
+)
+
+data_io = H5DataIO(data=np.random.randn(100, 100), maxshape=(None, 100))
+
+nwbfile.add_acquisition(
+    TimeSeries(
+        name="synthetic_timeseries",
+        description="Random values",
+        data=data_io,
+        unit="m",
+        rate=10e3,
+    )
+)
+
+with NWBHDF5IO("test_edit2.nwb", "w") as io:
+    io.write(nwbfile)
+
+##############################################
+# The ``None`` in ``maxshape`` means that the dataset has an unlimited shape. You can
+# also use an integer to specify a fixed ``maxshape``. If you do not specify a
+# ``maxshape``, then the dataset will have a fixed shape. You can change the shape of
+# this dataset.
+
+import h5py
+
+with h5py.File("test_edit2.nwb", "r+") as f:
+    f["acquisition"]["synthetic_timeseries"]["data"].resize((200, 100))
+
+##############################################
+# This will change the shape of the dataset in-place. If you try to change the shape of
+# a dataset with a fixed shape, you will get an error.
+#
+# Replacing a dataset in h5py
+# ----------------------------
+# There are several types of dataset edits that cannot be done in-place:
+#
+#     * Changing the shape of a dataset with a fixed shape
+#     * Changing the datatype of a dataset
+#     * Changing the compression of a dataset
+#     * Changing the chunking of a dataset
+#     * Changing the max-shape of a dataset
+#     * Changing the fill-value of a dataset
+#
+# For any of these, you will need to create a new dataset with the new shape, copying
+# the data from the old dataset to the new dataset, and deleting the old dataset.
+
+with h5py.File("test_edit2.nwb", "r+") as f:
+    data_values = f["acquisition"]["synthetic_timeseries"]["data"][:]
+    del f["acquisition"]["synthetic_timeseries"]["data"]
+    f["acquisition"]["synthetic_timeseries"].create_dataset(
+        name="data",
+        data=data_values,
+        maxshape=(None, 100),
+        chunks=(100, 100),
+        compression="gzip",
+        compression_opts=3,
+        fillvalue=0.0,
+        dtype=np.float64,
+    )
+
+##############################################
+# .. note::
+#   Because of the way HDF5 works, the ``del`` action will not actually free up any
+#   space in the HDF5 file. To free up space in the file, you will need to run the
+#   ``h5repack`` command line tool. See the `h5repack documentation
+#   <https://support.hdfgroup.org/HDF5/doc/RM/Tools.html#Tools-Repack>`_ for more
+#   information.
```

Comment:
`h5repack` just makes a new copy of the file. If we are creating a copy of the file anyways, then I think it may be easier to use the export functionality in PyNWB for this. @rly is this covered in any of the docs on export? I think the steps would be to:

- Open the file in PyNWB
- Replace the TimeSeries
- Use `NWBHDF5IO.export` to export to a new file.

Reply:
I can't recommend strongly enough against `h5repack` - I had to do some file fixes for an old Buzsaki project and the average write speed was ~4 MB/s and has no options to parallelize

You would absolutely get faster speeds by just 'reading' from the source NWB file and 'writing' a brand new one (which I assume is what `export` does)

Reply:
Huh, good to know